Hibernate added

Kukharrr opened the pull request.

Kukharrr committed:
Hibernate added

JXLEA commented on .mvn/wrapper/maven-wrapper.properties:
```diff
@@ -0,0 +1,19 @@
+# under the License.
+wrapperVersion=3.3.2
+distributionType=only-script
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.9.9/apache-maven-3.9.9-bin.zip
```
Pls, do not push configuration files, not good practice

JXLEA commented on docker-compose.yml:
```diff
@@ -0,0 +1,31 @@
+    ports:
+      - "8080:8080"
+
+#volumes:
```
for docker, it is a common practice to have /infrastructure directory, in which place docker file and docker-compose

JXLEA commented on src/main/java/com/example/epam/controller/TraineeController.java:
```diff
@@ -0,0 +1,80 @@
+
+    @PostMapping
+    public ResponseEntity<Trainee> createTrainee(@RequestBody TraineeCreateDto traineeCreateDto) {
+        System.out.println("registering new trainee");
```
pls, do not push System.out..., use Logger instead

JXLEA commented on src/main/java/com/example/epam/dao/TrainerDao.java:
```diff
@@ -0,0 +1,68 @@
+        try (Session session = sessionFactory.openSession()) {
+            session.beginTransaction();
+            session.persist(trainer);
+            session.getTransaction().commit();
```
pls, add transaction fallback behavior. 
In case you get an error during execution of the transaction what will be happening.

Kukharrr committed:
loggers added
added transaction fallback behavior
docker files are separated for now
System.out.println are not present in project for now(sorry for this, my bad)

Kukharrr committed:
@transactional now replaced with manual handling using commit and rollback
